OrbitLinuxTracing/GpuTracepointEventProcessorTest.cpp

```diff
@@ -0,0 +1,589 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "GpuTracepointEventProcessor.h"
+#include "OrbitLinuxTracing/TracerListener.h"
+#include "capture.pb.h"
+
+namespace LinuxTracing {
+
+namespace {
+
+class MockTracerListener : public TracerListener {
+ public:
+  MOCK_METHOD(void, OnSchedulingSlice, (orbit_grpc_protos::SchedulingSlice), (override));
+  MOCK_METHOD(void, OnCallstackSample, (orbit_grpc_protos::CallstackSample), (override));
+  MOCK_METHOD(void, OnFunctionCall, (orbit_grpc_protos::FunctionCall), (override));
+  MOCK_METHOD(void, OnIntrospectionScope, (orbit_grpc_protos::IntrospectionScope), (override));
+  MOCK_METHOD(void, OnGpuJob, (orbit_grpc_protos::GpuJob gpu_job), (override));
+  MOCK_METHOD(void, OnThreadName, (orbit_grpc_protos::ThreadName), (override));
+  MOCK_METHOD(void, OnThreadStateSlice, (orbit_grpc_protos::ThreadStateSlice), (override));
+  MOCK_METHOD(void, OnAddressInfo, (orbit_grpc_protos::AddressInfo), (override));
+  MOCK_METHOD(void, OnTracepointEvent, (orbit_grpc_protos::TracepointEvent), (override));
+  MOCK_METHOD(void, OnModulesUpdate, (orbit_grpc_protos::ModulesUpdateEvent), (override));
+};
+
+class GpuTracepointEventProcessorTest : public ::testing::Test {
+ protected:
+  void SetUp() override { processor_.SetListener(&mock_listener_); }
+
+  void TearDown() override {}
+
+  GpuTracepointEventProcessor processor_;
+  MockTracerListener mock_listener_;
+};
+
+AmdgpuCsIoctlPerfEvent MakeFakeAmdgpuCsIoctlPerfEvent(pid_t tid, uint64_t timestamp_ns,
+                                                      uint32_t context, uint32_t seqno,
+                                                      const std::string& timeline) {
+  AmdgpuCsIoctlPerfEvent event{
+      static_cast<uint32_t>(sizeof(amdgpu_cs_ioctl_tracepoint) + timeline.length() + 1)};
+  event.ring_buffer_record.sample_id.tid = tid;
+  event.ring_buffer_record.sample_id.time = timestamp_ns;
+  reinterpret_cast<amdgpu_cs_ioctl_tracepoint*>(event.tracepoint_data.get())->context = context;
+  EXPECT_EQ(event.GetContext(), context);
```

Comment:
This seems to test the `AmdgpuCsIoctrlPerfEvent`, which is not really related to the `GpuTracepointEventProcessor`, right? I am unsure if it is right to have those `EXPECT`s here. 
If this is more about to validate the results from the assignments after reinterpret_cast, CHECK might be the better choice. WDYT?

Reply:
You are right, all these `EXPECT` made me forget of `CHECK`. Done.